eva-pipeline/src/test/java/embl/ebi/variation/eva/pipeline/steps/VariantAnnotLoadBatchTest.java

```diff
@@ -24,17 +47,127 @@
  */
 public class VariantAnnotLoadBatchTest {
 
+
+    private VariantAnnotLoadBatch variantAnnotLoadBatch;
+    private ExecutionContext executionContext;
+
+    @Before
+    public void setUp() throws Exception {
+        executionContext = MetaDataInstanceFactory.createStepExecution().getExecutionContext();
+
+        variantAnnotLoadBatch = new VariantAnnotLoadBatch();
+    }
+
+    @Test
+    public void variantAnnotationReaderShouldReadAllFieldsInVepOutput() throws Exception {
+        VariantAnnotationLineMapper lineMapper = new VariantAnnotationLineMapper();
+        for (String annotLine : vepOutput.split("\n")) {
+            VariantAnnotation variantAnnotation = lineMapper.mapLine(annotLine, 0);
+            assertTrue(variantAnnotation.getConsequenceTypes() != null);
+        }
+    }
+
     @Test
-    public void variantAnnotationReaderShouldReadAllFieldsInVepOutput(){
+    public void variantAnnotationReaderShouldReadAllLinesInVepOutput() throws Exception {
+        String vepOutput = VariantAnnotLoadBatchTest.class.getResource("/annot.tsv.gz").getFile();
```

Comment:
Why you are using the file and not the vepOutput at the bottom of the test? Using the String you are not depending on external file and is specific for the test + easy to read (no need to open a file)
What about something like:
{code}
        ByteArrayOutputStream baostream = new ByteArrayOutputStream();
        OutputStream outStream = new GZIPOutputStream(baostream);
        outStream.write(vepOutput.getBytes("UTF-8"));
        outStream.close();
        byte[] compressedBytes = baostream.toByteArray();


        Resource r = new InputStreamResource(new GZIPInputStream(new ByteArrayInputStream(compressedBytes)));
        FlatFileItemReader<VariantAnnotation> reader = variantsAnnotLoadBatch.initReader(r);
{code}